projects/server-database/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"server-database/types"
+	"strconv"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/joho/godotenv"
+)
+
+func handleImages(w http.ResponseWriter, r *http.Request) {
+	conn, err := setDatabaseConnection()
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+	defer conn.Close(context.Background())
+	images, err := fetchImages(conn)
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+	indent := getIndentParam(r)
+	b, err := json.MarshalIndent(images, "", indent)
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(b)
+}
+
+func getIndentParam(r *http.Request) string {
+	params := r.URL.Query()
+
+	indentSize, err := strconv.Atoi(params.Get("indent"))
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+	indent := ""
+	for i := 0; i < indentSize; i++ {
+		indent += " "
+	}
+	return indent
+}
+
+func fetchImages(conn *pgx.Conn) ([]types.Image, error) {
+	var images []types.Image
+	rows, err := conn.Query(context.Background(), "SELECT title, url, alt_text FROM public.images")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: failed to fetch images: %v", err)
+	}
+	for rows.Next() {
+		var title, url, altText string
+		err = rows.Scan(&title, &url, &altText)
+		images = append(images, types.Image{Title: title, URL: url, AltText: altText})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: failed to scan image: %v", err)
+		}
+	}
+	return images, err
+}
+func setDatabaseConnection() (*pgx.Conn, error) {
+	databaseURL := os.Getenv("DATABASE_URL")
+	if databaseURL == "" {
+		fmt.Fprintln(os.Stderr, "DATABASE_URL is not set")
+		os.Exit(1)
+	}
+
+	conn, err := pgx.Connect(context.Background(), databaseURL)
```

Comment:
so, this works but in a real application with significant traffic it would be a big problem to create a new database connection for every query. it's costly to create and tear down connections (especially when one db has many application server instances connecting to it). 
Also, servers can run out of resources for creating connections - take a look at https://blog.cloudflare.com/how-to-stop-running-out-of-ephemeral-ports-and-start-to-love-long-lived-connections

So, to solve this problem, we typically reuse database connections using a connection pool. Think of this as like a lending library for DB connections - just as with a library, you borrow a book/connection, read it, and then give it back for the next person to use.

Take a look at https://pkg.go.dev/github.com/jackc/pgx/v4/pgxpool which is the commonly-used way of doing this for go/postgres.